woocommerce/wc-admin
# Add email address field to store details step in OBW

Fixes: #7515, #7518
Partially fixes: #7517

This PR does the following:

1. Adds loading indicator for store details step while data. This is because the profile data is loaded via AJAX and it is required prior setting up the forms.
1. Moves the setup client checkbox to Business Details step.
1. Adds `is_agree_marketing` and `store_email` to onboarding profile data. This is stored in the option `woocommerce_onboarding_profile`.
2. Validates the email field to be required if marketing checkbox is checked.
3. Validates the email on client side with a simple "@" character check and validates email field on the server-side using WordPress' built-in request argument validator `rest_validate_request_arg`.
1. Modify the relevant tracks.

Follow-ups intended in separate PRs:

1. Pre-fill email with jetpack and store email.
1. Show errors from the server-side (Currently if the email format is invalid, the server rejects the request but does not notify the user of the reason).
1. Possibly more robust email validation on client-side, but I reserve to not do this since it’s hard and we could possibly have a conflicting result between client and server sided validation.

### Accessibility




### Screenshots

![image](https://user-images.githubusercontent.com/3747241/130792445-e7509934-4c6a-493e-b207-a8fbc56c2f25.png)

![image](https://user-images.githubusercontent.com/3747241/130792391-052cc018-b4f5-4858-8383-95efe76c5c1b.png)

### Detailed test instructions:
- Start OBW and set up your browser console to monitor tracks. To do this, run `localStorage.setItem( 'debug', 'wc-admin:tracks' );`
- Observe "Get tips, product updates and inspiration straight to your mailbox" checkbox and "Email address" field in the Store Details step.
- Checking the checkbox should make the email field required, you should not be able to continue if it's not filled.
- Fill in the email address field with a valid email and click on continue.
- Observe in the track `wcadmin_storeprofiler_store_details_continue` with prop `email_signup` that appropriately flags if the user agreed to receive marketing emails.
- Continue until Business Features step.
- Observe the "I'm setting up a store for a client" checkbox in the step.
- Click on continue.
- Observe in the track `wcadmin_storeprofiler_store_business_details_continue_variant` with prop `setup_client` that appropriately flags if the user is setting up store for a client.
